controller/work_item_link_type.go

```diff
@@ -248,20 +255,22 @@ func (c *WorkItemLinkTypeController) Show(ctx *app.ShowWorkItemLinkTypeContext)
 			jerrors, httpStatusCode := jsonapi.ErrorToJSONAPIErrors(err)
 			return ctx.ResponseData.Service.Send(ctx.Context, httpStatusCode, jerrors)
 		}
-		// Convert the created link type entry into a rest representation
-		appLinkType := ConvertWorkItemLinkTypeFromModel(ctx.RequestData, *modelLinkType)
+		return ctx.ConditionalEntity(*modelLinkType, c.config.GetCacheControlWorkItemLinkTypes, func() error {
+			// Convert the created link type entry into a rest representation
+			appLinkType := ConvertWorkItemLinkTypeFromModel(ctx.RequestData, *modelLinkType)
 
-		// Enrich
-		hrefFunc := func(obj interface{}) string {
-			return fmt.Sprintf(app.WorkItemLinkTypeHref(spaceID, "%v"), obj)
-		}
-		linkCtx := newWorkItemLinkContext(ctx.Context, appl, c.db, ctx.RequestData, ctx.ResponseData, hrefFunc, nil)
-		err = enrichLinkTypeSingle(linkCtx, &appLinkType)
-		if err != nil {
-			jerrors, _ := jsonapi.ErrorToJSONAPIErrors(goa.ErrInternal("Failed to enrich link type: %s", err.Error()))
-			return ctx.InternalServerError(jerrors)
-		}
-		return ctx.OK(&appLinkType)
+			// Enrich
+			hrefFunc := func(obj interface{}) string {
+				return fmt.Sprintf(app.WorkItemLinkTypeHref(spaceID, "%v"), obj)
+			}
+			linkCtx := newWorkItemLinkContext(ctx.Context, appl, c.db, ctx.RequestData, ctx.ResponseData, hrefFunc, nil)
+			err = enrichLinkTypeSingle(linkCtx, &appLinkType)
+			if err != nil {
+				jerrors, _ := jsonapi.ErrorToJSONAPIErrors(goa.ErrInternal("Failed to enrich link type: %s", err.Error()))
```

Comment:
@aslakknutsen I applied the change and replace the same call in the utility functions of this file with a simple `return err` which will be converted in an JSON API error by the caller method.